agent/agentexec/exec.go

```diff
@@ -0,0 +1,86 @@
+package agentexec
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+	"strconv"
+
+	"golang.org/x/xerrors"
+)
+
+const (
+	// EnvProcPrioMgmt is the environment variable that determines whether
+	// we attempt to manage process CPU and OOM Killer priority.
+	EnvProcPrioMgmt  = "CODER_PROC_PRIO_MGMT"
+	EnvProcOOMScore  = "CODER_PROC_OOM_SCORE"
+	EnvProcNiceScore = "CODER_PROC_NICE_SCORE"
```

Comment:
Since these are agent-scoped they should probably start with `CODER_AGENT_`

Reply:
Actually, these should probably be turned into real flags/envs on the agent subcommand and then passed in to this function IMO

Reply:
I also wants to side step `serpent`, because we're `exec`'ing we want to remove as many opportunities as possible for someone to add something to `serpent` that may cause us to leak an open fd or something of that ilk

Reply:
I also wants to side step `serpent`, because we're `exec`'ing we want to remove as many opportunities as possible for someone to add something to `serpent` that may cause us to leak an open fd or something of that ilk

Reply:
I agree about the prefix but they already exist for the current implementation. Maybe we add new `CODER_AGENT` prefixed ones and deprecate the old ones.